### ISSUE TITLE
Improve version control and user experience

### ISSUE THREAD

**Issue by riban-bw** [zynthian/zynthian-issue-tracking#390]
**Is your feature request related to a problem? Please describe.**
Currently users are connected to a number of GitHub repositories which provide software updates. The default branch on each repository is _master_. Fixes are applie to this branch then users update to receive them. Feature and bug fix branches only receive significant testing if users are willing to configure their Zynthian to use the branch and this can be awkward if many branches exist.

**Describe the solution you'd like**
Add new branches called _stable_ and _testing_.  _master_ can be retired but needs to remain until all users have migrated from it. _master_ is updated to change the update scripts to use new branches.

A user should be connected to the _stable_ branch by default. Webconf basic view should offer a tickbox called, "Opt in to testing" (or similar) which toggles user between _stable_ and _testing_ branches of all repositoires. Advanced view in webconf can reveal individual repository selection drop-downs that offer all branches.

Developers work in _development_ branches and merge or pull request to _testing_ branches when satisfied development passes appropriate tests.

Testers subscribe to the _testing_ branch, validating functionality and reporting issues.

When _testing_ branch passes all tests and project manager is satisfied it is sufficiently stable it is merged to _stable_ branch. _stable_ branch is tagged as a release and stable image is created. (I don't see the advantage of nightly images except maybe in preparation for stable release.)

If an issue is reported against _stable_ release that is sufficiently significant to warrant an immediate fix then a fix is developed in a _devlopment_ branch (branched from _stable_) and tested then applied to _stable_ branch as a _hot fix_. The fix is also merged into testing. _stable_ branch is tagged, e.g. V1.1 and stable image is updated.

**Additional context**
This workflow provides a rolling but stable platform for performance users and a rolling testing platform for testers (similar to current _master_). It provides a fairly simple but robust method of development. There is no reduced functionality from current situation but improves user's and developer's understanding of the support mechanisms. It provides a simpler webconf interface for users to understand how to use the stable or testing versions whilst allowing power users / developers / testers simple access to other branches, e.g. to test a specific fix. Removal of _master_ may appease some users who may take offence to the use of this term. (I do not want to ignite a debate on this subject but we could take this opportunity to remove it from discussion.)

**Comment by riban-bw** [comment]
Toggling webconf tickbox would select _stable_ / _testing_ in each of the (hidden) repository drop-downs.

**Comment by riban-bw** [comment]
I have added stable and testing branches to relevant repositories.
I have created a zynthian-sys branch to test migration: _migrate_to_stable_branch_.
I have tested migration works pretty well.
I could submit a PR for this.

I have created a zynthian-webconf branch: _repository_mod_.
This has the changes to repository (although still called repository and on separate page to update - step by step...).
I also added indication of updates available to the dashboard.
I could submit a PR for this.

This can all be tested by ensuring zynthian is on master then selecting _migrate_to_stable_branch_ branch and performing an update. After the update all the repositories (exept sys) will show as stable.

Webconf changes can be tested by selecting _repository_mod_ branch and updating.
